src/controllers/publish.ts

```diff
@@ -579,13 +587,28 @@ export const uploadLookupTable = async (req: Request, res: Response, next: NextF
     }
     let errors: ViewError[] | undefined;
     const revisit = dataset.dimensions?.length > 0;
-
+    const dataPreview = await req.pubapi.getDimensionPreview(res.locals.dataset.id, dimension.id);
     if (req.method === 'POST') {
         logger.debug('User submitted a look up table');
         try {
             if (!req.file) {
-                logger.error('No file attached to request');
-                throw new Error('errors.csv.invalid');
+                res.status(400);
+                res.render('publish/upload-lookup', {
+                    ...dataPreview,
+                    revisit,
+                    errors: [
+                        {
+                            field: 'csv',
+                            message: {
+                                key: 'errors.upload.no_csv'
+                            }
+                        }
+                    ],
+                    uploadType: 'lookup',
+                    dimension,
+                    changeLookup: Boolean(dimension.type === 'lookup_table')
+                });
+                return;
```

Comment:
The error handling blocks here are very similar, could just throw an error with the necessary message and then to all the repeated stuff in the catch.

Reply:
They're similar but they are at points where you must bail out... E.g. no file... or in the case of the catch an error has occurred.  The one in the middle the 400 of course heads you down the "mismatch" path which is a different error page altogether.

Could reduce the code with a function if it its a big thing.